### PR TITLE
Improve exception handling and documentation of runtime attachment

### PR DESCRIPTION
**Pull request by jeanbisutti**
Improve exception handling and documentation of runtime attachment

### CODE REVIEW

File: runtime-attach/README.md [hunk 1]
```diff
@@ -4,12 +4,14 @@ If you can't update the JVM arguments to attach the [OpenTelemetry Java agent](h
...
 The `io.opentelemetry.contrib.attach.RuntimeAttach` class has an `attachJavaagentToCurrentJVM` method allowing to trigger the attachment of the OTel agent for Java.
 
-The attachment will not be initiated in the following cases:
+The attachment will _not_ be initiated in the following cases:
 * The `otel.javaagent.enabled` property is set to `false`
 * The `OTEL_JAVAAGENT_ENABLED` environment variable is set to `false`
 * The attachment is not requested from the _main_ thread
 * The attachment is not requested from the `public static void main(String[] args)` method
 * The agent is already attached
+* JRE (a JDK is necessary)
```
Comment (mateuszrzeszutek) [inline runtime-attach/README.md:13]:
```suggestion
* The application is running on a JRE (a JDK is necessary)
```

File: runtime-attach/README.md [hunk 2]
```diff
@@ -4,12 +4,14 @@ If you can't update the JVM arguments to attach the [OpenTelemetry Java agent](h
...
+* Read-only file system
```
Comment (mateuszrzeszutek) [inline runtime-attach/README.md:14]:
```suggestion
* The application is running on a read-only file system
```

**Comment by jeanbisutti** [comment]
Thank you for your suggestions @mateuszrzeszutek!

File: runtime-attach-core/src/main/java/io/opentelemetry/contrib/attach/CoreRuntimeAttach.java [hunk 3]
```diff
@@ -88,6 +93,24 @@ private static boolean agentIsDisabledWithEnvVar() {
     return "false".equals(agentEnabledEnvVarValue);
   }
 
+  private static void handleByteBuddyException(RuntimeException exception) {
+    handleNoAgentProvider(exception);
+    throw new RuntimeAttachException(
+        "A problem has appeared during the runtime attachment of the Java agent.", exception);
```
Comment (trask) [inline runtime-attach-core/src/main/java/io/opentelemetry/contrib/attach/CoreRuntimeAttach.java:99]:
```suggestion
        "A problem has occurred during the runtime attachment of the Java agent.", exception);
```

**Review by trask — APPROVED** [review]

**Comment by jeanbisutti** [comment]
The runtime attach files were moved in #385. So resynched with main and added a new commit.

**Pull request merged**